extensions/scarb-doc/src/types/crate_type.rs

```diff
@@ -61,9 +77,28 @@ impl Crate {
         db: &ScarbDocDatabase,
         module_ids: Vec<ModuleId>,
     ) -> &mut Module {
-        let mut current_module = &mut self.root_module;
-
-        for id in module_ids.iter() {
+        let mut current_module = {
+            if let Some(first_ancestor) = module_ids.first() {
+                if &self.root_module.module_id != first_ancestor {
+                    if let Some(index) = self
+                        .foreign_crates
+                        .iter_mut()
+                        .position(|module| module.module_id == *first_ancestor)
+                    {
+                        &mut self.foreign_crates[index]
+                    } else {
+                        self.foreign_crates
+                            .push(Module::new_virtual(db, *first_ancestor));
+                        &mut self.foreign_crates[0]
```

Comment:
Shouldn't this point to the last element, since push appends at the end of the collection? 🤔

Reply:
right, my bad 👍 👍